TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1156,6 +1159,281 @@ def exterior_boundaries(feature_layers, zoom,
+    # (combinatoric) set of all different levels.
+    for kind, features in admin_features.iteritems():
+        num_features = len(features)
+        envelopes = [g[0].envelope for g in features]
+
+        for i, feature in enumerate(features):
+            shape, props, fid = feature
+            envelope = envelopes[i]
+
+            # orient to ensure that the shape is to the
+            # left of the boundary.
+            boundary = orient(shape).boundary
+
+            # intersect with *preceding* features to remove
+            # those boundary parts. this ensures that there
+            # are no duplicate parts.
+            for j in range(0, i):
+                cut_shape, cut_props, cut_fid = features[j]
+                cut_envelope = envelopes[j]
+                if envelope.intersects(cut_envelope):
+                    boundary = _intersect_cut(boundary, cut_shape)[1]
+
+                if boundary.is_empty:
+                    break
+
+            # intersect with every *later* feature. now each
+            # intersection represents a section of boundary
+            # that we want to keep.
+            for j in range(i+1, num_features):
+                cut_shape, cut_props, cut_fid = features[j]
+                cut_envelope = envelopes[j]
+
+                if envelope.intersects(cut_envelope):
+                    inside, boundary = _intersect_cut(boundary, cut_shape)
+
+                    inside = _linemerge(inside)
+                    if not inside.is_empty:
+                        new_props = _merge_left_right_props(props, cut_props)
+                        new_props['id'] = props['id']
+                        _make_joined_name(new_props)
+                        new_features.append((inside, new_props, fid))
+
+                if boundary.is_empty:
+                    break
+
+            # anything left over at the end is still a boundary,
+            # but a one-sided boundary to international waters.
+            boundary = _linemerge(boundary)
+            if not boundary.is_empty:
+                new_props = props.copy()
+                _make_joined_name(new_props)
+                new_features.append((boundary, new_props, fid))
+
+
+    # use intracut for maritime, but it intersects in a positive
+    # way - it sets the tag on anything which intersects, whereas
+    # we want to set maritime where it _doesn't_ intersect. so
+    # we have to flip the attribute afterwards.
+    cutter = _Cutter(maritime_features, None,
+                     'maritime_boundary', 'maritime_boundary',
+                     2, _intersect_cut)
+
+    for shape, props, fid in new_features:
+        cutter.cut(shape, props, fid)
+
+    # flip the property, so define maritime_boundary=yes where
+    # it was previously unset and remove maritime_boundary=no.
+    for shape, props, fid in cutter.new_features:
+        maritime_boundary = props.pop('maritime_boundary', None)
+        if maritime_boundary is None:
```

Comment:
Ah, no. In the query, we're picking buffered land polygons with `maritime_boundary=no`, so the absence of `maritime_boundary` is when we want to add `maritime_boundary=yes`. How could I explain this better in the comment?